src/transformers/modeling_utils.py

```diff
@@ -2688,6 +2702,25 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                 }
             )
 
+            target_dtype = torch_dtype
+
+            if load_in_4bit:
+                _accelerate_has_custom_dtype = version.parse(importlib_metadata.version("accelerate")) > version.parse(
+                    "0.19.0"
+                )
+
+                if _accelerate_has_custom_dtype:
+                    from accelerate.utils import CustomDtype
+
+                    target_dtype = CustomDtype.INT4
+                else:
+                    logger.warning(
+                        "You are using `device_map='auto'` on a 4bit loaded version of the model. To automatically compute the appropriate device map, you should upgrade your `accelerate` library, `pip install --upgrade accelerare` or install it from source to support int4 auto device map calculation. You may encounter unexpected behavior, or pass your own device map"
+                    )
+                    target_dtype = torch.int8
```

Comment:
Here the user has indicated `load_in_4bit=True` with improper version of Accelerate, so we should throw an error instead of defaulting to 8bit.